core/gui/src/app/workspace/component/file-selection/file-selection.component.ts

```diff
@@ -29,25 +31,33 @@ export class FileSelectionComponent implements OnInit {
   ) {}
 
   ngOnInit() {
-    // if users already select some file, then show that selected dataset & related version
-    if (this.selectedFilePath && this.selectedFilePath !== "") {
-      const selectedDatasetFile = parseFilePathToDatasetFile(this.selectedFilePath);
-      this.selectedDataset = this.datasets.find(
-        d => d.ownerEmail === selectedDatasetFile.ownerEmail && d.dataset.name === selectedDatasetFile.datasetName
-      );
-      this.isDatasetSelected = !!this.selectedDataset;
+    // retrieve all the accessible datasets from the backend
+    this.datasetService
+      .retrieveAccessibleDatasets()
+      .pipe(untilDestroyed(this))
+      .subscribe(response => {
+        this.datasets = response.datasets;
+        this.isAccessibleDatasetsLoading = false;
 
-      if (this.selectedDataset && this.selectedDataset.dataset.did !== undefined) {
-        this.datasetService
-          .retrieveDatasetVersionList(this.selectedDataset.dataset.did)
-          .pipe(untilDestroyed(this))
-          .subscribe(versions => {
-            this.datasetVersions = versions;
-            this.selectedVersion = this.datasetVersions.find(v => v.name === selectedDatasetFile.versionName);
-            this.onVersionChange();
-          });
-      }
-    }
+        // if users already select some file, then ONLY show that selected dataset & related version
+        if (this.selectedFilePath && this.selectedFilePath !== "") {
```

Comment:
a general comment: to make it more readable, use early termination
```
if (xxx) { 
  return
}
// your logic here
```

also it's better if you can use pipe(filter()) to replace some of the if statements.

Reply:
changed